lib/Parse/ParsePattern.cpp

```diff
@@ -66,10 +66,22 @@ void Parser::DefaultArgumentInfo::setFunctionContext(
 
 static ParserStatus parseDefaultArgument(
     Parser &P, Parser::DefaultArgumentInfo *defaultArgs, unsigned argIndex,
-    Expr *&init, Parser::ParameterContextKind paramContext) {
+    Expr *&init, bool &hasInheritedDefaultArg,
+    Parser::ParameterContextKind paramContext) {
   SyntaxParsingContext DefaultArgContext(P.SyntaxContext,
                                          SyntaxKind::InitializerClause);
   SourceLoc equalLoc = P.consumeToken(tok::equal);
+  // Handle the special '= super' syntax for inherited default values in module
+  // interface files
+  if (P.SF.Kind == SourceFileKind::Interface) {
+    if (P.Tok.is(tok::kw_super) && P.peekToken().isAny(tok::comma, tok::r_paren)) {
```

Comment:
Can you expand this comment to explain that you're explicitly looking for `super,` and `super)`, and that future syntactic expansion around inherited default args would break this?